tests/test_flow/fakes.py

```diff
@@ -177,7 +212,13 @@ def create_aip_job(body, parent):
         # Emulate this behavior by setting the log level; this should filter out
         # all (or almost all) the logs. Tests that check the log output will
         # only test against logs which are printed locally.
-        with caplog.at_level(logging.CRITICAL):
+        #
+        # Additionally, AIP executions run in an isolated environment and do not
+        # have access to disk cache from local bionic instance. It must retrieve
+        # the data from GCS. Emulate this behavior by temporarily removing all
+        # files in the tmp_path directory. The files are restored after the
+        # AIP computation is completed.
+        with caplog.at_level(logging.CRITICAL), temporary_clean_dir(tmp_path):
             run_aip(body["trainingInput"]["args"][3])
```

Comment:
Do you need to call this function here? Would it be easier to run `body["trainingInput"]["args"]` as a shell command instead? That way, you won't need to use this private method here.

Reply:
If we run this in a separate process, we need to ensure that any writes to the GCS filesystem in the separate process is reflected here in the original process. If we have a fake GCS, then we need to pass a proxy of the fake GCS to the separate process. This is more involved.

Reply:
Ah, good point. I still think it's worthwhile and we should investigate if we can use our `PytestManager` to instantiate the fake gcs, but we don't have to do it in this PR.